src/YarpUtilities/include/BipedalLocomotionControllers/YarpUtilities/Helper.tpp

```diff
@@ -0,0 +1,238 @@
+    {
+        if (!(inputPtr->get(i).*YARP_UTILITES_GET_CHECKER_NAME(elementType))())
+        {
+            std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] "
+                         "The element of the list associated to the value named "
+                      << key << " is not a " << YARP_UTILITES_GET_ELEMENT_TYPE(elementType) << "."
+                      << std::endl;
+            return false;
+        }
+
+        vector[i] = convertValue<elementType>(inputPtr->get(i));
+    }
+    return true;
+}
+
+template <>
+bool getVectorFromSearchable<std::vector<bool>>(const yarp::os::Searchable& config,
+                                                const std::string& key,
+                                                std::vector<bool>& vector)
+{
+
+    static_assert(YARP_UTILITES_CHECK_ELEMENT_SUPPORT(bool),
+                  "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                  "function getElementFromSearchable() cannot be called with the desired "
+                  "element type");
+
+    yarp::os::Value* value;
+    if (!config.check(key, value))
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] "
+                     "Missing field "
+                  << key << std::endl;
+        return false;
+    }
+
+    if (value->isNull())
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] Empty "
+                     "input value named "
+                  << key << std::endl;
+        return false;
+    }
+
+    if (!value->isList())
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                     "value named "
+                  << key << "is not associated to a list." << std::endl;
+        return false;
+    }
+
+    yarp::os::Bottle* inputPtr = value->asList();
+    if (inputPtr == nullptr)
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                     "list associated to the value named "
+                  << key << " is empty." << std::endl;
+        return false;
+    }
+
+    // resize the vector
+    vector.resize(inputPtr->size());
+
+    for (int i = 0; i < inputPtr->size(); i++)
+    {
+        if (!(inputPtr->get(i).isBool()) && !(inputPtr->get(i).isInt()))
+        {
+            std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] "
+                         "The element of the list associated to the value named "
+                      << key << " is not a boolean ." << std::endl;
+            return false;
+        }
+
+        vector[i] = convertValue<bool>(inputPtr->get(i));
+    }
+    return true;
+}
+
+template <typename T> void mergeSigVector(yarp::sig::Vector& vector, const T& t)
+{
+    for (int i = 0; i < t.size(); i++)
+        vector.push_back(t(i));
```

Comment:
Maybe you can add a specialization for std containers where you use iterators instead of iterating over the index.

Reply:
Maybe it is also worth checking that the data of the t vector [is_convertible](https://en.cppreference.com/w/cpp/types/is_convertible) to ``double``.

Reply:
```cpp
template <typename T>
void mergeSigVector<std::vector<T>>(yarp::sig::Vector& vector, const std::vector<T>& t);
```
cannot specialize 
```cpp
template <typename T> 
void mergeSigVector(yarp::sig::Vector& vector, const T& t);
```
becouse is a **partial** specialization that is not supported by C++ (please check [here](https://stackoverflow.com/questions/38199641/template-specialization-on-stdvectort).
 
For this reason, I overload the `mergeSigVector` as
```cpp
template <typename T>
void mergeSigVector(yarp::sig::Vector& vector, const std::vector<T>& t);
```
 The overall outcome is the same

Reply:
Maybe it is worth doing the same with all the sequence containers (https://en.cppreference.com/w/cpp/container). I wonder whether there exists a way (maybe using SFINAE) to have it working with all the containers which have ``begin`` and ``end`` and the pointed value is a double. Maybe a possible solution could be something similar to the example in https://en.cppreference.com/w/cpp/types/void_t.

Reply:
I solved the problem in this way:
1. I used the `operator[]` to access the element of the vector
2. I used `SFINAE` technique in `mergeSigVector()` function to handle both scalar and vector input 

Commit: https://github.com/dic-iit/bipedal-locomotion-controllers/pull/5/commits/a096bfb3a24009384231308d0ce20ced69160279

Reply:
Notice that some containers don't have the ``operator[]`` like the [``list``](https://en.cppreference.com/w/cpp/container/list) or the [``set``](https://en.cppreference.com/w/cpp/container/set). 

You can decide whether this could be useful or not. I thought it would have been interesting to be independent from the container as long as they provide an iterator.

Reply:
In https://github.com/dic-iit/bipedal-locomotion-controllers/pull/5/commits/f0b4611da6207ae46a04f54f7769f637537109ca,  I handled different types of `std` container in `YarpUtilities::mergeSigVector()`. 
This will require C++17 @traversaro

Reply:
Awesome!